cli/azd/internal/repository/infra_confirm.go

```diff
@@ -207,10 +171,10 @@ func (i *Initializer) promptDbName(ctx context.Context, database appdetect.Datab
 		}
 
 		if strings.ContainsAny(dbName, " ") {
-			i.console.MessageUxItem(ctx, &ux.WarningMessage{
+			console.MessageUxItem(ctx, &ux.WarningMessage{
 				Description: "Database name contains whitespace. This might not be allowed by the database server.",
 			})
-			confirm, err := i.console.Confirm(ctx, input.ConsoleOptions{
+			confirm, err := console.Confirm(ctx, input.ConsoleOptions{
 				Message: fmt.Sprintf("Continue with name '%s'?", dbName),
```

Comment:
Looks like the `input-text + warning + confirm` pattern is repeated multiple times around here. Consider making it on general function or even better, supported by input.Console.

Reply:
See the answer [here](https://github.com/Azure/azure-dev/pull/4509#discussion_r1828181015)